### PR TITLE
CHK-1438-rrn-into-template

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes
Added RRN handling in the template of the success email.

#### Motivation and Context
RRN handling has been added in the template of the success email, retrieved from the authorization completion event.

#### How Has This Been Tested?
The tests were performed locally.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/UserReceiptMailBuilder.kt [hunk 1]
```diff
@@ -109,8 +109,8 @@ class UserReceiptMailBuilder(@Autowired private val confidentialMailUtils: Confi
             PspTemplate(
               transactionAuthorizationRequestData.pspBusinessName,
               FeeTemplate(amountToHumanReadableString(transactionAuthorizationRequestData.fee))),
-            transactionAuthorizationRequestData
-              .authorizationRequestId, // FIXME: put here RRN received from PGS
+            Optional.ofNullable(transactionAuthorizationCompletedData.rrn)
+              .orElse(transactionAuthorizationRequestData.authorizationRequestId),
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/UserReceiptMailBuilder.kt:113]:
```suggestion
            transactionAuthorizationCompletedData.rrn?:
              transactionAuthorizationRequestData.authorizationRequestId,
```
Replace with elvis operator

File: src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/UserReceiptMailBuilderTest.kt [hunk 2]
```diff
@@ -83,7 +83,8 @@ class UserReceiptMailBuilderTest {
               dateString,
               totalAmountWithFeeString,
               PspTemplate(TransactionTestUtils.PSP_BUSINESS_NAME, FeeTemplate(feeString)),
-              baseTransaction.transactionAuthorizationRequestData.authorizationRequestId,
+              Optional.ofNullable(baseTransaction.transactionAuthorizationCompletedData.rrn)
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/UserReceiptMailBuilderTest.kt:86]: Use Elvis operator here

File: src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/UserReceiptMailBuilderTest.kt [hunk 3]
```diff
@@ -83,7 +83,8 @@ class UserReceiptMailBuilderTest {
...
+                .orElse(baseTransaction.transactionAuthorizationRequestData.authorizationRequestId),
               baseTransaction.transactionAuthorizationCompletedData.authorizationCode,
               PaymentMethodTemplate(
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/UserReceiptMailBuilderTest.kt:89]: Add a test for both a transaction with and without RRN field valued with assertion on sent rrn value to notifications service

**Review by pietro-tota — COMMENTED** [review]
Just some comments

**Review by infantesimone — APPROVED** [review]

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**